### DISCUSSION TITLE
`req` object passed to `provider.authorize()` is not compatible with `getToken({ req })`

### DISCUSSION THREAD

**Discussion by iMrDJAi** [nextauthjs/next-auth#4640] [Help]
### Provider type

Credentials

### Environment

"next-auth": "^4.3.4"

### Reproduction URL

null

### Describe the issue

`req` object passed to `provider.authorize()` is not compatible with `getToken({ req })`. Only query, body, headers and method properties are available, causing `getToken({ req })` to return `null`.
https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/routes/callback.ts#L334-L339

### How to reproduce

```ts
CredentialsProvider({
  id: 'verify-email',
  credentials: {
    token: {}
  },
  async authorize(form = {} as any, req) {
    const sessionToken = await getToken({ req })
    console.log(sessionToken) // null
    return true
  }
})
```

### Expected behavior

`getToken({ req })` should return the token object properly.

**Comment by balazsorban44** [comment]
What is the use case exactly here? You can use https://next-auth.js.org/configuration/initialization#advanced-initialization if you have a valid case.

**Reply by PaulNext** [reply]
> What is the use case exactly here? You can use https://next-auth.js.org/configuration/initialization#advanced-initialization if you have a valid case.

Yes, you can use advanced features like this in API routing, but where do you get the authOptions in the front-end unstable_getServerSession?
// login.tsx
export async function getServerSideProps(context) {
  const session = await unstable_getServerSession(context.req, context.res,authOptions)
  if (session) {
    return {
      redirect: { 
        destination: "/xxx" 
      }
    };
  }
  return {
    props: {
      session
    }
  };
}
// [...nextauth].ts
export default async function auth(req: NextApiRequest, res: NextApiResponse) {
return await NextAuth(req, res,{})
}

**Comment by iMrDJAi** [comment]
@balazsorban44 The use case is not relevant here. The problem is it's impossible to use `jwt.getToken()` to obtain the token object from the request object inside `authorize()`. This is useful to check if the email verification token matches the logged in user in my case.

**Comment by iMrDJAi** [comment]
@balazsorban44 I believe that this should be labeled as an enchantment instead of a question.

**Comment by iMrDJAi** [comment]
Related: https://github.com/nextauthjs/next-auth/discussions/4647
